internal/summon/command/temp_factory.go

```diff
@@ -63,24 +63,33 @@ func defaultTempPath(sharedMemoryDir string) string {
 }
 
 // Push creates a temp file with given value. Returns the path.
-func (tf *TempFactory) Push(value string) string {
-	f, _ := ioutil.TempFile(tf.path, ".summon")
-	defer f.Close()
+func (tf *TempFactory) Push(value string) (string, error) {
+	f, err := ioutil.TempFile(tf.path, ".summon")
+	defer func() {
+		_ = f.Close()
+	}()
+
+	if err != nil {
+		return "", err
+	}
+
+	_, err = f.Write([]byte(value))
+	if err != nil {
+		return "", err
+	}
 
-	f.Write([]byte(value))
 	name := f.Name()
 	tf.files = append(tf.files, name)
-	return name
+	return name, nil
 }
 
 // Cleanup removes the temporary files created with this factory.
 func (tf *TempFactory) Cleanup() {
 	for _, file := range tf.files {
-		os.Remove(file)
+		_ = os.Remove(file)
 	}
 	// Also remove the tempdir if it's not the shared memory directory
 	if !strings.Contains(tf.path, defaultSharedMemoryDir) {
-		os.Remove(tf.path)
+		_ = os.Remove(tf.path)
 	}
-	tf = nil
```

Comment:
Two things here:
- If we're not clearing the object (which I agree is weird), we need to at least reset `tf.files[]` to an empty array so that repeated invocation won't try to delete already-deleted files. 
- If we're not clearing the object here too, it is a bit problematic that we remove the directory in cleanup since it allows for calling `Push()` again but the directory won't be there.

Reply:
There is no such thing as clearing an object in Go as far as I know. But `tf.files = nil` is a worthwhile thing.

Reply:
> `tf.files = nil` is a worthwhile thing.

This may need to be a new empty array though instead of nil